appcelerator/titanium_mobile
# fix(android): TabGroup crashes if tab "title" property not set as of 8.0.2

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-27188

**Summary:**
- [TIMOB-27188](https://jira.appcelerator.org/browse/TIMOB-27188) Fixed bug where `TabGroup` will crash if "title" property is `null` or left undefined.
  * This is a regression as of Titanium 8.0.2.
- Added additional `null` checks to `TabGroup` Java code.
- Added icon-only tab group unit tests for top/bottom styles.

**Test:**
1. Create a Classic "Default Project".
2. Replace project's "app.js" with the below code.
3. Build and run on Android.
4. Verify 2 tabs appear at the top without titles; only icons. _(This used to crash.)_
5. Tap on the right tab.
6. In the log, verify the following was printed: `Tab 1 was unselected`
7. In the log, verify the following was printed: `Tab 2 was selected`
8. In the code below, uncomment the `tabGroupSettings.style` line.
9. Rebuild and run on Android.
10. Verify 2 tabs appears at the bottom without titles; only icons. _(This used to crash.)_
11. Tap on the right tab.
12. In the log, verify the following was printed: `Tab 1 was unselected`
13. In the log, verify the following was printed: `Tab 2 was selected`
14. In the code below, uncomment the `title: title` line.
15. Rebuild and run on Android.
16. Verify that the bottom tabs have both titles and icons.
17. In the code below, comment-out (or delete) the `tabGroupSettings.style` line.
18. Rebuild and run on Android.
19. Verify that the top tabs have both titles and icons.

```javascript
function createTab(title, iconPath) {
	var window = Ti.UI.createWindow({ title: title });
	window.add(Ti.UI.createLabel({ text: title + " View" }));
	var tab = Ti.UI.createTab({
//		title: title,
		icon: iconPath,
		window: window,
	});
	tab.addEventListener("unselected", function(e) {
		Ti.API.info("@@@ " + title + " was unselected.");
	});
	tab.addEventListener("selected", function(e) {
		Ti.API.info("@@@ " + title + " was selected.");
	});
	return tab;
}

var tabGroupSettings = {
	tabs: [
		createTab("Tab 1", "/assets/images/tab1.png"),
		createTab("Tab 2", "/assets/images/tab2.png")
	],
};
if (Ti.Android) {
//	tabGroupSettings.style = Ti.UI.Android.TABS_STYLE_BOTTOM_NAVIGATION;
}
var tabGroup = Ti.UI.createTabGroup(tabGroupSettings);
tabGroup.open();
```
